### DISCUSSION TITLE
is it possible to show image of molecule and its 3D in one jupyter cell output?

### DISCUSSION THREAD

**Discussion by khoroshyy** [rdkit/rdkit#4950] [Q&A] [answered]
So that is the question. When I show 3d structure the image shown before disappears. Is there a workarround? 
Thanks.

**Comment by dehaenw** [comment]
Hello @khoroshyy 
How are you displaying the image? Using just "display" I have no problem rendering a mol and then its 3d structure. if i run the below cell in jupyter:
```python
import rdkit
from rdkit import Chem
from rdkit.Chem import AllChem
mol = Chem.MolFromSmiles("c1ccccc1c1ccccc1") #biphenyl
display(mol)
mol = Chem.AddHs(mol)
AllChem.EmbedMolecule(mol)
display(mol)mo
```
i get the expected ouput:
![image](https://user-images.githubusercontent.com/66372095/151159570-d8ad8981-9127-4c2f-aff6-b1db52d27e94.png)
Perhaps you are rendering by just having a line saying "mol" (or whatever the name of your mol is) or MolToImage(mol) - in that case only the last instance in the cell gets rendered. Putting display() around it fixes it in either case

**Reply by khoroshyy** [reply]
Hi, thanks for your reply. 
What I wanted is to show 2d structure and 3d interactive structure as an output of one cell.
I can show 2d and 3d separately.
![separate](https://user-images.githubusercontent.com/1734704/151811384-00bbed90-1e6f-4e87-8626-36b4067ca8c7.png)
But if I try to show them together only interactive  3d is shown
![image](https://user-images.githubusercontent.com/1734704/151811684-80aa67e2-1044-4e80-8f28-13beff90b4a3.png)

I guess it is a limitation of the 3d widget, but maybe it is possible to overcome.

**Comment by greglandrum** [comment, marked as answer]
Hi @khoroshyy, it's no doubt possible to do this with some coding (for example by combining the 3D widget an the molecule image in a table), but there's no built-in way to do it

**Reply by khoroshyy** [reply]
@greglandrum Thanks. Could you point me out a bit in the direction of the widget combination?

**Reply by greglandrum** [reply]
That ended up being easier than I thought it was going to be. The trick is to use jupyter's `display()` function:
https://gist.github.com/greglandrum/7bc0489e30e9ccec8f6af6b86a5bb993
